src/main/java/lotto/domain/Lotto.java

```diff
@@ -1,87 +1,56 @@
 package lotto.domain;
 
+import lotto.generator.NumberGenerator;
+
 import java.util.*;
-import java.util.stream.Collectors;
 
 public class Lotto {
     private static final int LOTTO_MAX_SOCKET = 6;
-    private static final String LOTTO_WINNING_NUMBER_SPLIT_KEYWORD = ",";
 
-    private Set<LottoNumber> lottoNumbers;
+    private final Set<LottoNumber> lottoNumbers;
 
-    public static Lotto newAutomatic() {
-        return new Lotto();
+    public static Lotto automatic(NumberGenerator numberGenerator) {
+        return new Lotto(numberGenerator);
     }
 
-    public static Lotto newManual(Set<LottoNumber> lottoNumbers) {
+    public static Lotto manual(Set<LottoNumber> lottoNumbers) {
         return new Lotto(lottoNumbers);
     }
 
-    public static Lotto newManual(List<LottoNumber> lottoNumbers) {
-        return new Lotto(lottoNumbers.stream().collect(Collectors.toSet()));
-    }
-
-    public static Lotto newManual(String lottoNumbers) {
-        return new Lotto(convertToLotto(lottoNumbers));
+    public static Lotto manual(List<Integer> lottoNumbers) {
+        return new Lotto(changeToLottoNumbers(lottoNumbers));
     }
 
-    private Lotto() {
-        generatorLottoNumbers();
-        sortLottorNumbers();
+    private Lotto(NumberGenerator numberGenerator) {
+        this(generatorLottoNumbers(numberGenerator));
     }
 
     private Lotto(Set<LottoNumber> lottoNumbers) {
         validate(lottoNumbers);
-        this.lottoNumbers = lottoNumbers;
-        sortLottorNumbers();
+        this.lottoNumbers = new TreeSet<>(lottoNumbers);
     }
 
     public Set<LottoNumber> getLottoNumbers() {
-        return this.lottoNumbers;
+        return new TreeSet<>(this.lottoNumbers);
     }
 
     public boolean isExistNumber(LottoNumber lottoNumber) {
-        return lottoNumbers.contains(lottoNumber);
+        return this.lottoNumbers.contains(lottoNumber);
     }
 
     public int getMatchedCount(Lotto lotto) {
-        int matchedCount = 0;
-        for (LottoNumber lottoNumber : lotto.getLottoNumbers()) {
-            matchedCount += countingMatched(lottoNumber);
-        }
-        return matchedCount;
+        return (int) this.getLottoNumbers().stream()
```

Comment:
아래 코멘트에 대해 고민해 봐요.
https://github.com/next-step/java-lotto/pull/438#discussion_r402377995

```suggestion
        return (int) this.lottoNumbers.stream()
```